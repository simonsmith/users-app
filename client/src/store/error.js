import {createReducer} from 'redux-starter-kit';

const initialState = {
  message: null,
  request: null,
};

export const errorReducer = createReducer(initialState, {
  API_REQUEST_FAILURE: (state, action) => {
    return {
      message: action.payload.toString(),
      response: action.payload.response,
    };
  },
});
