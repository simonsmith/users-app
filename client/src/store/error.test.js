import {errorReducer} from './error';

describe('reducer', () => {
  test('handle no state being passed', () => {
    expect(errorReducer(undefined, {})).toEqual({
      message: null,
      request: null,
    });
  });

  test('stores the error and response', () => {
    const error = new Error('oh no');
    error.response = {
      foo: 'bar',
    };
    const afterState = errorReducer(undefined, {
      payload: error,
      type: 'API_REQUEST_FAILURE',
    });

    expect(afterState).toMatchSnapshot();
  });
});
