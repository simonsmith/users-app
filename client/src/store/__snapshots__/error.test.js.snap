// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer stores the error and response 1`] = `
Object {
  "message": "Error: oh no",
  "response": Object {
    "foo": "bar",
  },
}
`;
